tls/s2n_connection.c

```diff
@@ -601,6 +543,12 @@ int s2n_connection_wipe(struct s2n_connection *conn)
     struct s2n_connection_hash_handles hash_handles = {0};
     struct s2n_connection_hmac_handles hmac_handles = {0};
 
+    struct s2n_handshake_hashes *handshake_hashes = conn->handshake.hashes;
+    if (!handshake_hashes) {
```

Comment:
I'm curious, why is this the behavior that we want? If they've freed the hashes, should we be re-allocating them?

Reply:
Yes. s2n_connection_wipe is called in order to reuse a connection. In order to reuse a connection, it needs handshake hashes.

We do a similar same thing with the handshake io buffer: we free it after the handshake, then re-allocate it on a new connection.

Reply:
to test my understanding, we are calling `s2n_handshake_hashes_new` to allocate the memory if it didnt already exist.

curious if we test the connection being re-used scenario and if we specifically test this scenario?

Reply:
Correct. And we do test that scenario :) https://github.com/aws/s2n-tls/pull/2987#discussion_r681232202